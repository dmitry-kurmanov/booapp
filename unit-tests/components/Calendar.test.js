import { Calendar } from './testbuild.js';

test('monthNumber', () => {
	const calendar = new Calendar({
		target: document.body,
		props: {},
	});
	const currentMonth = new Date().getMonth();

	expect(calendar.monthNumber).toEqual(currentMonth);
});

test('yearNumber', () => {
	const calendar = new Calendar({
		target: document.body,
		props: {},
	});
	const currentYear = new Date().getFullYear();

	expect(calendar.yearNumber).toEqual(currentYear);
});


test('goToNextMonth', () => {
	const calendar = new Calendar({
		target: document.body,
		props: {},
	});
	
	const date = new Date();
	const currentMonth = date.getMonth();
	const currentYear = date.getFullYear();

	calendar.goToNextMonth();
	expect(calendar.yearNumber).toEqual(currentYear);
	expect(calendar.monthNumber).toEqual(currentMonth + 1);

	calendar.monthNumber = 11;

	calendar.goToNextMonth();
	expect(calendar.yearNumber).toEqual(currentYear + 1);
	expect(calendar.monthNumber).toEqual(0);
});

test('goToPrevMonth', () => {
	const calendar = new Calendar({
		target: document.body,
		props: {},
	});
	
	const date = new Date();
	const currentMonth = date.getMonth();
	const currentYear = date.getFullYear();

	calendar.goToPrevMonth();
	expect(calendar.yearNumber).toEqual(currentYear);
	expect(calendar.monthNumber).toEqual(currentMonth - 1);

	calendar.monthNumber = 0;

	calendar.goToPrevMonth();
	expect(calendar.yearNumber).toEqual(currentYear - 1);
	expect(calendar.monthNumber).toEqual(11);
});
