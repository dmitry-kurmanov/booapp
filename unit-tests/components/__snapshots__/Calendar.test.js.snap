// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="booapp-calendar"
>
  <div
    class="booapp-calendar__header"
  >
    <button
      class="booapp-calendar__prev-month-btn"
    >
      ← Prev
    </button>
     
    <div
      class="booapp-calendar__header-info"
    >
      April
       
      2021
    </div>
     
    <button
      class="booapp-calendar__next-month-btn"
    >
      → Next
    </button>
  </div>
   
  <div
    class="booapp-calendar__weekdays"
  >
    <div
      class="booapp-calendar__weekday"
    >
      Sun
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Mon
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Tue
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Wed
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Thu
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Fri
       
    </div>
    <div
      class="booapp-calendar__weekday"
    >
      Sat
       
    </div>
  </div>
   
  <div
    class="booapp-calendar__days"
  >
    <div
      class="booapp-calendar__day booapp-calendar__day--prev-month booapp-calendar__day--weekend"
    >
      28
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--prev-month"
    >
      29
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--prev-month"
    >
      30
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--prev-month"
    >
      31
    </div>
     
    <div
      class="booapp-calendar__day"
    >
      1
    </div>
    <div
      class="booapp-calendar__day"
    >
      2
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      3
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      4
    </div>
    <div
      class="booapp-calendar__day"
    >
      5
    </div>
    <div
      class="booapp-calendar__day"
    >
      6
    </div>
    <div
      class="booapp-calendar__day"
    >
      7
    </div>
    <div
      class="booapp-calendar__day"
    >
      8
    </div>
    <div
      class="booapp-calendar__day"
    >
      9
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      10
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      11
    </div>
    <div
      class="booapp-calendar__day"
    >
      12
    </div>
    <div
      class="booapp-calendar__day"
    >
      13
    </div>
    <div
      class="booapp-calendar__day"
    >
      14
    </div>
    <div
      class="booapp-calendar__day"
    >
      15
    </div>
    <div
      class="booapp-calendar__day"
    >
      16
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      17
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      18
    </div>
    <div
      class="booapp-calendar__day"
    >
      19
    </div>
    <div
      class="booapp-calendar__day"
    >
      20
    </div>
    <div
      class="booapp-calendar__day"
    >
      21
    </div>
    <div
      class="booapp-calendar__day"
    >
      22
    </div>
    <div
      class="booapp-calendar__day"
    >
      23
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      24
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--weekend"
    >
      25
    </div>
    <div
      class="booapp-calendar__day"
    >
      26
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--today"
    >
      27
    </div>
    <div
      class="booapp-calendar__day"
    >
      28
    </div>
    <div
      class="booapp-calendar__day"
    >
      29
    </div>
    <div
      class="booapp-calendar__day"
    >
      30
    </div>
     
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month booapp-calendar__day--weekend"
    >
      1
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month booapp-calendar__day--weekend"
    >
      2
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month"
    >
      3
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month"
    >
      4
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month"
    >
      5
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month"
    >
      6
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month"
    >
      7
       
    </div>
    <div
      class="booapp-calendar__day booapp-calendar__day--next-month booapp-calendar__day--weekend"
    >
      8
       
    </div>
  </div>
</div>
`;
